modules/core/04-channel/types/upgrade.go

```diff
@@ -69,3 +72,22 @@ func (uf UpgradeFields) ValidateBasic() error {
 func (ut UpgradeTimeout) IsValid() bool {
 	return !ut.Height.IsZero() || ut.Timestamp != 0
 }
+
+// HasPassed returns true if the upgrade has passed the timeout height or timestamp
+func (ut UpgradeTimeout) HasPassed(ctx sdk.Context) (bool, string) {
+	if !ut.IsValid() {
+		return true, "upgrade timeout cannot be empty"
+	}
```

Comment:
I think we can remove this. We should validate a timeout before using `HasPassed`

Reply:
is there any harm in keeping this check though? otherwise we end up returning a false negative in the event the timeout validation doesn't occur.